### PR TITLE
fix(query): parse date in dst return err

### LINKED ISSUE #12098: bug: to_date cause session panic
**Summary**

Under time zone Asia/Shanghai:

```
mysql> select to_date('1941-03-15');
ERROR 2013 (HY000): Lost connection to MySQL server during query
No connection. Trying to reconnect...
Connection id:    15
Current database: *** NONE ***

ERROR 2013 (HY000): Lost connection to MySQL server during query
No connection. Trying to reconnect...
Connection id:    16
Current database: *** NONE ***

ERROR 2013 (HY000): Lost connection to MySQL server during query

```

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

date 1941-03-15 with timezone Asia/Shanghai 

In main branch:

read_date_text will parse date string => NaivDate => Datetime<Tz>

Shanghai under DST begin with 1941-03-15 00:00:00. So will panic with No such local time.

```rust
impl<T: fmt::Debug> LocalResult<T> {
    /// Returns the single unique conversion result, or panics accordingly.
    pub fn unwrap(self) -> T {
        match self {
            LocalResult::None => panic!("No such local time"),
            LocalResult::Single(t) => t,
            LocalResult::Ambiguous(t1, t2) => {
                panic!("Ambiguous local time, ranging from {:?} to {:?}", t1, t2)
            }
        }
    }
}
```

- Closes #12098

### CODE REVIEW

File: src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs [hunk 1]
```diff
@@ -32,7 +32,11 @@ use crate::cursor_ext::cursor_read_bytes_ext::ReadBytesExt;
 
 pub trait BufferReadDateTimeExt {
     fn read_date_text(&mut self, tz: &Tz) -> Result<NaiveDate>;
-    fn read_timestamp_text(&mut self, tz: &Tz) -> Result<DateTime<Tz>>;
+    fn read_timestamp_text(
+        &mut self,
+        tz: &Tz,
+        only_date_text: bool,
+    ) -> Result<(Option<DateTime<Tz>>, Option<NaiveDate>)>;
```
Comment (youngsofun) [inline src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs:39]: should add an enum?

Comment (TCeason) [reply src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs:39]: Will fix it later

File: src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs [hunk 2]
```diff
@@ -98,21 +111,18 @@ where T: AsRef<[u8]>
...
+                tz.from_utc_datetime(
                     &NaiveDate::from_ymd_opt(1000, 1, 1)
                         .unwrap()
                         .and_hms_opt(0, 0, 0)
                         .unwrap(),
-                ))
+                )
             } else {
-                Ok(dt)
+                dt
```
Comment (youngsofun) [inline src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs:125]: maybe we can split read_timestamp_text into 2 local functions: date part and time part, read_date_text only call the former

Comment (youngsofun) [reply src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs:125]: so we nolonger need `.0` or choose DateTime from enum?

Comment (TCeason) [reply src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs:125]: > maybe we can split read_timestamp_text into 2 local functions: date part and time part, read_date_text only call the former

Can't do this. Support to_date(timestamp)

File: src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs (hunk 1 above)
Comment (TCeason) [reply src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs]: Fixed

**Review by youngsofun — APPROVED** [review]

**Review by sundy-li — APPROVED** [review]

**Pull request merged**